File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        n = len(nums)

        total = sum(nums)
        target = total - x

        if target < 0: return -1
        if target == 0: return n

        left = 0
        current_sum = 0
        maxLength = -1

        for right in range(n):
            current_sum += nums[right]

            while left <= right and current_sum > target:
                current_sum -= nums[left]
                left += 1

            if current_sum == target:
                maxLength = max(maxLength, right - left + 1)

        return -1 if maxLength == -1 else n - maxLength

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.js
/**
 * @param {number[]} nums
 * @param {number} x
 * @return {number}
 */
var minOperations = function(nums, x) {
    const n = nums.length;

    let total = 0;
    for (let i = 0; i < n; i++) total += nums[i];

    const target = total - x;

    if (target < 0) return -1;
    if (target === 0) return n;

    let left = 0;
    let sum = 0;
    let maxLength = -1;

    for (let right = 0; right < n; right++) {
        sum += nums[right];

        while (left <= right && sum > target) {
            sum -= nums[left++];
        }

        if (sum === target) {
            maxLength = Math.max(maxLength, right - left + 1);
        }
    }

    return maxLength === -1 ? -1 : n - maxLength;
};
